hypothesis-python/tests/py3/test_lookup.py

```diff
@@ -599,3 +599,55 @@ def test_bytestring_is_valid_sequence_of_int_and_parent_classes(type_):
         st.from_type(typing.Sequence[type_]),
         lambda val: isinstance(val, typing.ByteString),
     )
+
+
+def supports_protocol(protocol, inst):
+    try:
+        # in python 3.8 this works
+        return isinstance(inst, protocol)
+    except TypeError:
+        # in python < 3.8 Protocols cannot be used with isinstance
+        # so we do this which is less nice...
+        return issubclass(type(inst), protocol)
+
+
+def supports_casting(typ, thing):
+    try:
+        typ(thing)
+        return True
+    # if anything bad happened here return False
+    except Exception:
+        return False
+
+
+@pytest.mark.parametrize("protocol", [typing.SupportsAbs, typing.SupportsRound])
+@given(data=st.data())
+def test_supportsop_types_support_protocol(protocol, data):
+    # test values drawn from SupportsOp types are indeed considered instances
+    # of that type.
+    value = data.draw(st.from_type(protocol))
+    # check that we aren't somehow generating instances of the protocol itself
+    assert value.__class__ != protocol
+    assert supports_protocol(protocol, value)
+
+
+@pytest.mark.parametrize(
+    "protocol, typ",
+    [
+        (typing.SupportsFloat, float),
+        (typing.SupportsInt, int),
+        (typing.SupportsBytes, bytes),  # noqa: B1
+        (typing.SupportsComplex, complex),
+    ],
+)
+@given(data=st.data())
+def test_supportscast_types_support_protocol_or_are_castable(protocol, typ, data):
+
+    value = data.draw(st.from_type(protocol))
+    # check that we aren't somehow generating instances of the protocol itself
+    assert value.__class__ != protocol
+    try:
+        # test values drawn from the protocol types are
```

Comment:
Are what?

I'd also consider writing this as a single assertion - `assert issubclass(type(value), protocol) or supports_casting(typ, value)`.  Otherwise an internal assertion error could be hidden by the `except` block!

Reply:
oh! yeah nice catch